crashtracker/src/crash_handler.rs

```diff
@@ -73,14 +74,29 @@ fn make_receiver(
         "{}",
         serde_json::to_string(&config)?
     )?;
-    writeln!(
-        receiver.stdin.as_ref().unwrap(),
-        "{}",
-        serde_json::to_string(&metadata)?
-    )?;
+
+    update_metadata(metadata)?;
     Ok(receiver)
 }
 
+/// Updates the crashtracker metadata for this process
+/// Metadata is stored in a global variable and sent to the crashtracking
+/// receiver when a crash occurs.
+///
+/// PRECONDITIONS:
+///     None
+/// SAFETY:
+///     Crash-tracking functions are not reentrant.
+///     No other crash-handler functions should be called concurrently.
+/// ATOMICITY:
+///     This function is not atomic. A crash during its execution may lead to
+///     unexpected crash-handling behaviour.
```

Comment:
Should the crash tracker maybe "suspend" itself or otherwise take care not to avoid issues here? E.g. in runtimes where multiple threads may be running, we can't guarantee that a crash won't happen in the middle of this update.

Reply:
Replaced with an atomic variable